src/iop/lens.cc

```diff
@@ -2098,8 +2098,13 @@ static void modflags_changed(GtkWidget *widget, gpointer user_data)
 void gui_changed(dt_iop_module_t *self, GtkWidget *w, void *previous)
 {
   dt_iop_lensfun_params_t *p = (dt_iop_lensfun_params_t *)self->params;
+  dt_iop_lensfun_gui_data_t *g = (dt_iop_lensfun_gui_data_t *)self->gui_data;
 
-  if(p->tca_r != 1.0 || p->tca_b != 1.0) p->tca_override = 1;
+  // update gui to show/hide tca sliders if tca_override was changed
+  if(w == g->tca_override)
+  {
+    gui_update(self);
```

Comment:
The preferred pattern is to call gui_changed from gui_update (with NULL widget) if any gui elements need to be adjusted (made invisible etc) after a value change and make the adjustments here. For a simple example see globaltonemap (or others).

Reply:
Makes sense. I modified the code. It got slightly more complex, but should be fine now.